certbot/main.py

```diff
@@ -215,6 +215,10 @@ def _find_cert(config, domains, certname):
               RenewableCert instance or None.
     """
     action, lineage = _find_lineage_for_domains_and_certname(config, domains, certname)
+    if action == "renew":
```

Comment:
@ohemorange my instinct is that we should land this as-is. We could consider whether it's sensible to have a wishlist refactor ticket for "move all stateful information out of `config` into another object", but the set of things we'd need to move would be larger than this one variable -- it would I think include every config variable that the user can set interactively and which persists for renewal (installer, authenticator, webroot map, etc)